std/bitmanip.d

```diff
@@ -1513,6 +1653,67 @@ public:
         }
     }
 
+    /**
+     * Creates `BitArray` from a string.
+     * Params:
+     *   s = string consisting of `0`, `1` and separators `_`.
+     * Throws:
+     *   `Exception` if the given string contains invalid characters.
+     */
+    static BitArray fromString(string s) pure @system
+    {
+        import std.exception : enforce;
+        import std.string : lastIndexOfNeither;
+        import std.algorithm.searching : count;
+
+        enforce(s.lastIndexOfNeither("01_") == -1,
+            "String contains invalid characters, only '0', '1', '_' are allowed.");
+
+        BitArray ba;
+        ba.length = s.length - s.count('_');
+        size_t i = 0;
+        foreach (c; s)
+        {
+            if (c == '_')
```

Comment:
Making these if statements into a `switch` will speed this up. Make sure the `0` case is explicit and not covered by `default`.